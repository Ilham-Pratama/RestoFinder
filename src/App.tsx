import React from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { StatusBar, useColorScheme } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';

import HomeScreen from './screens/HomeScreen';
import DetailScreen from './screens/DetailScreen';
import SearchModalScreen from 'screens/SearchModal';
import SearchCityModal from 'screens/SearchCityModal';

import { RootStackParamList } from './types/globalTypes';
import { AppDarkTheme, AppLightTheme } from 'shared/themes';
import getRandomCityName from 'shared/libs/getRandomCity';

const Stack = createNativeStackNavigator<RootStackParamList>();

export const queryClient = new QueryClient();

export const Navigations = ({
  initialRouteName = 'Home',
  fixedInitialCity
}: INavigations) => {
  return (
    <Stack.Navigator initialRouteName={initialRouteName}>
      <Stack.Group>
        <Stack.Screen
          name="Home"
          component={HomeScreen}
          options={{ title: 'Home' }}
          initialParams={{
            term: '',
            location: fixedInitialCity
              ? 'New York, New York'
              : getRandomCityName(),
            categories: []
          }}
        />
        <Stack.Screen
          name="Details"
          component={DetailScreen}
          options={({ route }) => ({
            title: route.params.name
          })}
        />
      </Stack.Group>
      <Stack.Group screenOptions={{ presentation: 'modal' }}>
        <Stack.Screen
          name="SearchModal"
          component={SearchModalScreen}
          options={{ title: 'Search Restaurants' }}
          initialParams={{
            term: '',
            location: '',
            categories: []
          }}
        />
        <Stack.Screen
          name="SearchCityModal"
          component={SearchCityModal}
          initialParams={{
            term: '',
            location: '',
            categories: []
          }}
          options={{ title: 'Search Location' }}
        />
      </Stack.Group>
    </Stack.Navigator>
  );
};

const App: React.FC = () => {
  const isDarkMode = useColorScheme() === 'dark';
  return (
    <QueryClientProvider client={queryClient}>
      <NavigationContainer theme={isDarkMode ? AppDarkTheme : AppLightTheme}>
        <Navigations />
      </NavigationContainer>
      <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
    </QueryClientProvider>
  );
};

interface INavigations {
  initialRouteName?: keyof RootStackParamList;
  fixedInitialCity?: boolean;
}

export default App;
