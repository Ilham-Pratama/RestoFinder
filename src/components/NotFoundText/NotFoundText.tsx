import React from 'react';
import { View } from 'react-native';
import sharedStyles from 'shared/styles';
import ThemedText from '../ThemedText';

export const DEFAULT_NOT_FOUND_TEXT = "Oops, couldn't find anything ðŸ˜¢";

interface INotFoundText {
  text?: string;
}

const NotFoundText = ({ text = DEFAULT_NOT_FOUND_TEXT }: INotFoundText) => {
  return (
    <View
      accessibilityRole="text"
      style={[sharedStyles.alignItemsCenter, sharedStyles.marginTop1]}>
      <ThemedText style={[sharedStyles.marginVertical3]} color="gray">
        {text}
      </ThemedText>
    </View>
  );
};

export default NotFoundText;
