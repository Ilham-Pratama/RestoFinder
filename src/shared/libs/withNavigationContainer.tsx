import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { AppLightTheme, AppDarkTheme } from 'shared/themes';

const withNavigationContainer = (component: any, darkMode: boolean = false) => {
  return () => (
    <NavigationContainer theme={darkMode ? AppDarkTheme : AppLightTheme}>
      {component}
    </NavigationContainer>
  );
};

export default withNavigationContainer;
