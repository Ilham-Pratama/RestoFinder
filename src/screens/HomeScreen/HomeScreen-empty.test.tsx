import React from 'react';

// Note: test renderer must be required after react-native.
import { render } from '@testing-library/react-native';
import { Navigations } from 'src/App';
import withNavigationContainer from 'shared/libs/withNavigationContainer';
import { DEFAULT_NOT_FOUND_TEXT } from 'components/NotFoundText/NotFoundText';

jest.mock('shared/api/useBusinessList', () => () => ({
  data: {
    total: 0,
    businesses: []
  },
  isLoading: false,
  isError: false
}));

it('renders and assert loading state', () => {
  const Component = withNavigationContainer(
    <Navigations initialRouteName="Home" fixedInitialCity />
  );

  const component = render(<Component />);

  expect(component.toJSON()).toMatchSnapshot();

  expect(component.getByText('Grab your')).toBeTruthy();
  expect(component.getByText('delicious meal!')).toBeTruthy();

  expect(component.getByText('Top Restaurants').props.style[0]).toHaveLength(3);

  expect(component.getByText(DEFAULT_NOT_FOUND_TEXT)).toBeTruthy();
});
