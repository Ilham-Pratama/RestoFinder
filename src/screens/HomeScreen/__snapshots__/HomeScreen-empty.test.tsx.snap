// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and assert loading state 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="#FFF"
        color="#c3f229"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Welcome"
        titleColor="#1C1C1E"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "#F2F2F2",
            },
            undefined,
          ]
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessibilityRole="header"
            style={
              Array [
                Object {
                  "marginHorizontal": 20,
                },
                Object {
                  "marginVertical": 10,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 45,
                  },
                  Object {
                    "color": "#1C1C1E",
                  },
                ]
              }
            >
              Grab your
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 45,
                    "fontWeight": "bold",
                  },
                  Object {
                    "color": "#1C1C1E",
                  },
                ]
              }
            >
              delicious meal!
            </Text>
          </View>
          <View
            accessibilityRole="search"
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 30,
                  "elevation": 3,
                  "flexDirection": "row",
                  "marginHorizontal": 20,
                  "paddingHorizontal": 12,
                  "shadowColor": "black",
                  "shadowOffset": Object {
                    "height": 5,
                    "width": 5,
                  },
                  "shadowOpacity": 0.1,
                }
              }
            >
              <RNSVGSvgView
                accessibilityRole="image"
                align="xMidYMid"
                bbHeight={20}
                bbWidth={20}
                fill="#000"
                focusable={false}
                height={20}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                vbHeight={512}
                vbWidth={512}
                width={20}
              >
                <RNSVGGroup
                  fill={4278190080}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M500.3 443.7l-119.7-119.7c27.22-40.41 40.65-90.9 33.46-144.7C401.8 87.79 326.8 13.32 235.2 1.723C99.01-15.51-15.51 99.01 1.724 235.2c11.6 91.64 86.08 166.7 177.6 178.9c53.8 7.189 104.3-6.236 144.7-33.46l119.7 119.7c15.62 15.62 40.95 15.62 56.57 0C515.9 484.7 515.9 459.3 500.3 443.7zM79.1 208c0-70.58 57.42-128 128-128s128 57.42 128 128c0 70.58-57.42 128-128 128S79.1 278.6 79.1 208z"
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                accessibilityRole="text"
                style={
                  Array [
                    Object {
                      "marginLeft": 15,
                      "marginVertical": 15,
                      "paddingRight": 10,
                    },
                    Object {
                      "color": "black",
                    },
                  ]
                }
              >
                New York, New York
              </Text>
            </View>
          </View>
          <Text
            style={
              Array [
                Array [
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                  Object {
                    "marginTop": 10,
                  },
                  Object {
                    "marginHorizontal": 20,
                  },
                ],
                Object {
                  "color": "#1C1C1E",
                },
              ]
            }
          >
            Top Restaurants
          </Text>
          <View
            accessibilityRole="text"
            style={
              Array [
                Object {
                  "alignItems": "center",
                },
                Object {
                  "marginTop": 5,
                },
              ]
            }
          >
            <Text
              color="gray"
              style={
                Array [
                  Array [
                    Object {
                      "marginVertical": 15,
                    },
                  ],
                  Object {
                    "color": "gray",
                  },
                ]
              }
            >
              Oops, couldn't find anything ðŸ˜¢
            </Text>
          </View>
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
